ci/do_ci.sh

```diff
@@ -214,6 +215,44 @@ if [[ "$CI_TARGET" == "bazel.release" ]]; then
   echo "bazel contrib release build..."
   bazel_contrib_binary_build release
 
+  exit 0
+elif [[ "$CI_TARGET" == "bazel.distribution" ]]; then
+  setup_clang_toolchain
+
+  if ! gpg --list-secret-keys "*"; then
```

Comment:
for this im wondering about moving it into the `envoy.gpg.sign` python lib - we can add a flag `--add-key` for non-release CI

Reply:
i have implemented the required bits to move this into `envoy.gpg.sign` but when i tried running this under `bazel build` it fails to create the key

i have raised an upstream (`python-gnupg`) bug here https://bitbucket.org/vinay.sajip/python-gnupg/issues/161/gen_key-fails-when-run-under-bazel-build

Reply:
resolved